### PR TITLE
Add geoviews offline features and tiles docs

### PR DESCRIPTION
**Pull request by ahuang11**
Directly based on https://discourse.pangeo.io/t/geoviews-offline/3487/2
Partially based on https://discourse.holoviz.org/t/using-geoviews-tile-sources-offline/4859/5

Planning to do tiles soon.

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -75,15 +79,82 @@ in how non-Python packages are built between conda-forge and defaults.
...
+
+Please ensure to set ``resources=INLINE`` if the machine you're using is completely
+offline and you intend to view the output on that machine.
+Failure to do so will result in the HTML file appearing empty when opened.
+
+If you wish to change the default data directory, follow these steps:
+
+1. Create a new directory and move the data::
+
+    import cartopy
```
Comment (Hoxbro) [inline doc/index.rst:129]: You need pathlib import here.

**Comment by Hoxbro** [comment]
Maybe think about if we should move this information into another place than the front page.

**Comment by ahuang11** [comment]
Yeah I was uncertain on whether I should put this on front page or one of the gallery items because of
```
1. Create a new cartopy environment (or use an existing one)::

    conda create -n cartopy_env python=3.10

2. Install the required packages::

    conda install -c conda-forge geoviews cartopy cartopy_offlinedata
```

**Comment by Hoxbro** [comment]
I think this should have a new page. I would build it up like a how-to, e.g, `How to use tiles offline` and put it in the user guide for now.

**Comment by ahuang11** [comment]
image
image

**Comment by Hoxbro** [comment]
I will look into this more tomorrow, but I think cache_tiles could be added directly to geoviews.util.

**Comment by ahuang11** [comment]
Yeah that sounds like a good idea.

I'm not exactly sure how `WMTS` works under the hood; does it use cartopy or bokeh  to fetch the tiles underneath?
```


class WMTS(_GeoFeature):
    """
    The WMTS Element represents a Web Map Tile Service specified as a
    URL containing {x}, {y}, and {z} templating variables, e.g.:

    https://maps.wikimedia.org/osm-intl/{Z}/{X}/{Y}@2x.png
    """

    crs = param.ClassSelector(default=ccrs.GOOGLE_MERCATOR, class_=ccrs.CRS, doc="""
        Cartopy coordinate-reference-system specifying the
        coordinate system of the data. Inferred automatically
        when _Element wraps cartopy Feature object.""")

    group = param.String(default='WMTS')

    layer = param.String(doc="The layer on the tile service")

    def __init__(self, data, kdims=None, vdims=None, **params):
        if ((MercatorTileSource and isinstance(data, MercatorTileSource)) or
            (GoogleTiles and isinstance(data, GoogleTiles))):
            data = data.url
        elif WebMapTileService and isinstance(data, WebMapTileService):
            pass
        elif not isinstance(data, str):
            raise TypeError(
                f'{type(self).__name__} data should be a tile service '
                f'URL not a {type(data).__name__} type.'
            )
        super().__init__(data, kdims=kdims, vdims=vdims, **params)

    def __call__(self, *args, **kwargs):
        return self.opts(*args, **kwargs)

```

**Comment by ahuang11** [comment]
But yeah, maybe this caching thing could be used to fix
https://github.com/holoviz/geoviews/issues/650

**Comment by ahuang11** [comment]
Any other thoughts on this PR?

**Review by Hoxbro — COMMENTED** [review]
The additions look fine. 

Maybe have some reference to external resources if users want to know more. E.g., one to `cartopy_offline` and one to Bokeh with resources `INLINE`. 

In the future, we will need to convert this to real markdown files, as we have done with Panel.

File: doc/user_guide/Using_Features_Offline.ipynb [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    "    conda create -n cartopy_env python=3.10\n",
+    "    ```\n",
+    "\n",
+    "2. Install the required packages:\n",
+    "\n",
+    "    ```bash\n",
+    "    conda install -c conda-forge geoviews cartopy cartopy_offlinedata\n",
+    "    ```\n",
+    "\n",
+    "    Or if you have an environment already, you may just need `cartopy_offlinedata`:\n",
```
Comment (ahuang11) [inline doc/user_guide/Using_Features_Offline.ipynb:26]:
```suggestion
    "    Or if you have an environment already, you may just need [`cartopy_offlinedata`](https://anaconda.org/conda-forge/cartopy_offlinedata):\n",
```

File: doc/user_guide/Using_Features_Offline.ipynb [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    "    gv.extension(\"bokeh\")\n",
+    "\n",
+    "    coastline = gv.feature.coastline()\n",
+    "    borders = gv.feature.borders()\n",
+    "    world = (coastline * borders).opts(global_extent=True)\n",
+    "\n",
+    "    gv.save(world, \"world.html\", resources=INLINE)\n",
+    "    ```\n",
+    "\n",
+    "    Please ensure to set ``resources=INLINE`` if the machine you're using is completely\n",
```
Comment (ahuang11) [inline doc/user_guide/Using_Features_Offline.ipynb:62]:
```suggestion
    "    Please ensure to set [`resources=INLINE`](https://docs.bokeh.org/en/latest/docs/reference/resources.html#bokeh.resources.INLINE) if the machine you're using is completely\n",
```

**Comment by Hoxbro** [comment]
> In the future, we will need to convert this to real markdown files, as we have done with Panel.

I just tried, and this is already possible. So the notebooks can be converted to Markdown files.

**Comment by ahuang11** [comment]
How do I convert the notebooks to Markdown files?

**Comment by Hoxbro** [comment]
I just copied the content to a new markdown file.

File: doc/user_guide/Using_Features_Offline.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+Now, we will verify that the shapefiles are available offline.
+
+1. Ensure offline shapefiles were downloaded:
+
+    ```python
+    from pathlib import Path
+    import cartopy
+
+    data_dir = Path(cartopy.config["pre_existing_data_dir"])
```
Comment (Hoxbro) [inline doc/user_guide/Using_Features_Offline.md:36]: I needed to use `cartopy.config["data_dir"]` to find some files. Also, make some sense with the environment variable set later.

File: doc/user_guide/Using_Features_Offline.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    conda create -n cartopy_env python=3.10
+    ```
+
+2. Install the required packages:
+
+    ```bash
+    conda install -c conda-forge geoviews cartopy cartopy_offlinedata
+    ```
+
+    Or if you have an environment already, you may just need [`cartopy_offlinedata`](https://anaconda.org/conda-forge/cartopy_offlinedata):
```
Comment (Hoxbro) [inline doc/user_guide/Using_Features_Offline.md:20]: I would make a note that this package is pretty big ~200MB.

File: doc/user_guide/Using_Features_Offline.md (hunk 4 above)
Comment (ahuang11) [reply doc/user_guide/Using_Features_Offline.md:36]: This should be the file directory where the cartopy_offlinedata was downloaded, so I'm wondering which files exactly are you missing?

Comment (Hoxbro) [reply doc/user_guide/Using_Features_Offline.md:36]: It works now for me, but I have no idea why :upside_down_face:

**Review by Hoxbro — APPROVED** [review]

**Comment by ahuang11** [comment]
Thanks for reviewing!

**Pull request merged**